gobblin-data-management/src/main/java/gobblin/data/management/copy/replication/ReplicationDataValidPathPicker.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.copy.replication;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Ordering;
+
+import gobblin.data.management.version.TimestampedDatasetVersion;
+import gobblin.data.management.version.finder.AbstractDatasetVersionFinder;
+import gobblin.data.management.version.finder.DateTimeDatasetVersionFinder;
+import gobblin.data.management.version.finder.GlobModTimeDatasetVersionFinder;
+import gobblin.dataset.FileSystemDataset;
+
+/**
+ * Used to pick the valid Paths for data replication based on {@link #ReplicationDataRetentionCategory.Type}
+ * @author mitu
+ *
+ */
+public class ReplicationDataValidPathPicker {
+
+  public static Collection<Path> getValidPaths(HadoopFsEndPoint hadoopFsEndPoint) throws IOException{
```

Comment:
The current implementation supports a few hand-picked DatasetVersionFinders and selection policies for the ones to be copied.

The retention management supports many different DatasetVersionFinders (and new ones can be added) plus different selection policies (top K,  before, after date, etc.) It is worthwhile to talk to @pcadabam and see how we can use those in the case of replication in a more generic manner.